sirepo/package_data/static/js/raydata.js

```diff
@@ -30,70 +33,88 @@ SIREPO.app.factory('raydataService', function(appState, requestSender, simulatio
 	return cols.length > 0 ? [firstColHeading].concat(cols) : [];
     };
 
-    self.getScansInfo = function(scanUuids, successCallback, options) {
-	function maybeGetOnlyOneScan(scans) {
-	    return s ? scans[0] : scans;
-	}
+    self.getScansInfo = function(successCallback, options) {
+	function helper(successcallback, options, recursionDepth) {
+	    const s = Object.keys(appState.models.scans.selected);
+	    if (s.every((e) => {
+		// POSIT: If start is present so are the other fields we need
+		return e in (simulationDataCache.scans || {}) && simulationDataCache.scans[e].start;
+	    })) {
+		successCallback(
+		    s.map(
+			u => simulationDataCache.scans[u]
+		    ).sort((a, b) => a.start - b.start),
+		    simulationDataCache.scanInfoTableCols
+		);
+		return;
+	    }
 
-	let s = false;
-	if (typeof scanUuids === 'string') {
-	    scanUuids = [scanUuids];
-	    s = true;
-	} else if (! Array.isArray(scanUuids)) {
-	    scanUuids = Object.keys(scanUuids);
-	}
-	if (scanUuids.every((e) => {
-	    // POSIT: If start is present so are the other fields we need
-	    return e in (simulationDataCache.scans || {}) && simulationDataCache.scans[e].start;
-	})) {
-	    successCallback(
-		maybeGetOnlyOneScan(
-		    scanUuids.map((u) => simulationDataCache.scans[u]
-		).sort((a, b) => a.start > b.start ? 1 : -1)),
-		simulationDataCache.scanInfoTableCols
+	    if (recursionDepth > 0) {
+		throw new Error(`infinite recursion detected scans=${JSON.stringify(s)} cache=${JSON.stringify(simulationDataCache.scans)}`);
+	    }
+	    requestSender.sendStatelessCompute(
+		appState,
+		(json) => {
+		    self.updateScansInCache(json.data.scans);
+		    self.updateScanInfoTableColsInCache(json.data.cols);
+		    self.getScansInfo(successCallback, options, recursionDepth + 1);
+		},
+		{
+		    method: 'scan_info',
+		    scans: s,
+		},
+		options
 	    );
-	    return;
 	}
-	requestSender.sendStatelessCompute(
-	    appState,
-	    (json) => {
-		successCallback(
-		    maybeGetOnlyOneScan(self.updateScansInCache(json.data.scans)),
-		    self.updateScanInfoTableColsInCache(json.data.cols)
-		);
-	    },
-	    {
-		method: 'scan_info',
-		scans: scanUuids,
-	    },
-	    options
-	);
+	helper(successCallback, options, 0);
+    };
+
+    self.getScansRequestPayload = function(scanUuids) {
+	return (scanUuids || Object.keys(appState.models.scans.selected)).map(s => {
+	    return {
+		models: appState.models,
+		report: s,
+		simulationType: SIREPO.APP_SCHEMA.simulationType,
+		simulationId: appState.models.simulation.simulationId,
+	    };
+	});
     };
 
     self.nextPngImageId = function() {
 	return 'raydata-png-image-' + (++id);
     };
 
-    self.parseComputeModel = function(computeModel) {
-	const p = computeModel.split('animation');
-	if (p.length !== 2) {
-	    throw new Error(`cannot parse computeModel=${computeModel}`);
+    self.selectOrDeselectScan = function(scan) {
+	if (! ('selected' in scan)) {
+	    scan.selected = false;
```

Comment:
Can you give me some insight into how I could make it clearer?

What a scan get's selected for the first time (either through user clicking the checkbox or because we are notified of it through polling) then it won't have the 'selected' key so we add it and set it to false. But the following line will set it to true which is what we really want.

Reply:
```
if (selected !== undefined) {
    scan.selected = selected;
}
else if ('selected' in scan) {
    scan.selected = !scan.selected;
}
else {
    scan.selected = true;
}
```

Reply:
a2236021f thanks